src/backend/cuda/jit.cpp

```diff
@@ -446,10 +449,10 @@ static char *irToPtx(string IR, size_t *ptx_size)
 
     NVVM_CHECK(nvvmGetCompiledResultSize(prog, ptx_size), "Can not get ptx size");
 
-    char *ptx = new char[*ptx_size];
-    NVVM_CHECK(nvvmGetCompiledResult(prog, ptx), "Can not get ptx from NVVM IR");
+    unique_ptr<char[]> ptx{new char[*ptx_size]};
+    NVVM_CHECK(nvvmGetCompiledResult(prog, ptx.get()), "Can not get ptx from NVVM IR");
     NVVM_CHECK(nvvmDestroyProgram(&prog), "Failed to destroy program");
-    return ptx;
+    return ptx.release();
```

Comment:
Is this necessary? Can't we return a `unique_ptr` instead of returning a `char *`?